File: Completed/JavaScript/559.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number}
 */
var maxDepth = function(root) {
  if (root === null) return 0
  let depth = 1
  for(let i = 0 ; i < root.children.length;i++) {
   let c = root.children[i]
   depth = Math.max(depth, 1 + maxDepth(c))
  }
  return depth
};

File: Completed/Go/559.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

 func maxDepth(root *Node) int {
    if root == nil {
        return 0
    }
    
    t := 0
    for _, child := range root.Children {
        res := maxDepth(child)
        if res > t {
            t = res
        }
    }
    
    return t+1
}